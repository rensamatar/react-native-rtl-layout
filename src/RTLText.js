import _ from 'lodash'
import React, { Component } from 'react'
import { Text, View, StyleSheet } from 'react-native'

export default class RTLText extends Component {
  render() {
    let style = StyleSheet.flatten(this.props.style || {})
    let locale = this.props.locale == 'ar' || null
    let fontSize = _.get(style, 'fontSize', this.props.fontSize || 18)
    let textAlign = (locale && locale == 'ar' ? 'right' : 'left')
    style = _.extend(style, { fontSize: fontSize })
    return (
      <Text
        {...this.props}
        textAlign={textAlign}
        style={style}>
        {this.props.children}
      </Text>
    )
  }
}