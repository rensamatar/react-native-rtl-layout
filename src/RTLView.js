import _ from 'lodash'
import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'

export default class RTLView extends Component {

  render() {
    let style = StyleSheet.flatten(this.props.style || {})
    let locale = this.props.locale == 'ar' || null
    let flexDirection = _.get(style, 'flexDirection', 'row')
    if (locale && flexDirection == 'row') { flexDirection = 'row-reverse' }
    else if (locale && flexDirection == 'row-reverse') { flexDirection = 'row' }
    style = _.extend(style, { flexDirection: flexDirection })

    return (
      <View
        {...this.props}
        style={style}>
        {this.props.children}
      </View>
    )
  }
}