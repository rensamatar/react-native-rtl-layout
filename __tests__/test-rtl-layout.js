import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import RTLView from '../src/RTLView'
import RTLText from '../src/RTLText'

it('renders correctly', () => {
  const tree = renderer.create(
    <RTLView />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly', () => {
  const tree = renderer.create(
    <RTLText />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})