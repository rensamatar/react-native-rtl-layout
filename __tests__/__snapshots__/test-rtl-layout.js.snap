// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
/>
`;

exports[`renders correctly 2`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "fontSize": 18,
    }
  }
  textAlign="left"
/>
`;
